internal/ast/expression.go

```diff
@@ -722,8 +741,9 @@ func (expr *Expression) Evaluate(scope *Package) error {
 		err = errors.New("unknown expression type")
 	}
 
-	if err == nil {
-		expr.EvaluationState = EvaluationStateComplete
+	if err != nil {
+		return fmt.Errorf("evaluate %+v: %w", expr, err)
```

Comment:
What about just using the string representation of the expression here?
```suggestion
		return fmt.Errorf("evaluate %q: %w", expr.Text, err)
```